app/src/main/java/com/example/tasky/agenda_details/presentation/components/Photos.kt

```diff
@@ -39,19 +60,102 @@ fun EmptyPhotos() {
                 horizontalArrangement = Arrangement.Center
             ) {
                 Icon(
-                    Icons.Filled.Add,
-                    modifier = Modifier.size(30.dp),
+                    painter = painterResource(id = R.drawable.ic_add),
+                    modifier = Modifier.size(15.dp),
                     tint = colorResource(id = R.color.gray),
                     contentDescription = null
                 )
-                Spacer(modifier = Modifier.padding(5.dp))
+                Spacer(modifier = Modifier.padding(10.dp))
                 Text(
-                    text = "Add Photos",
+                    text = stringResource(R.string.add_photos),
                     fontSize = 16.sp,
                     fontWeight = FontWeight.SemiBold,
                     color = colorResource(id = R.color.gray)
                 )
             }
         }
     }
+}
+
+@Composable
+fun Photos(
+    selectedPhotoUris: List<Uri>,
+    compressedImages: List<Bitmap?>,
+    compressAndAddImage: (Context, List<Uri>) -> Unit,
+    photoSkipCount: Int,
+    onAddPhotosClick: () -> Unit
+) {
+    val context = LocalContext.current
+
+    LaunchedEffect(selectedPhotoUris) {
+        compressAndAddImage(context, selectedPhotoUris)
+    }
```

Comment:
Doesn't feel very efficient to me to always compress **all** images even if you add or remove just one. Why not just compress the one you added and leave the rest unchanged?

Also, could it make sense to move this call into the ViewModel, since this doesn't really need to get triggered from the UI, right? The ViewModel knows, when the Uris change.